### ISSUE TITLE
unoconv PDF: unix process error: wait for unix process: exit status 5

### ISSUE THREAD

**Issue by PierreSchwang** [gotenberg/gotenberg#516]
When using gotenberg:7 and the endpoint `forms/chromium/convert/html` with the additional multipart data `pdfFormat=PDF/A-1a` the following error ocurred (Convert HTML to PDF/A-1a):
```json
{"level":"error","ts":1664889966.8094878,"logger":"api","msg":"convert HTML to PDF: convert PDF: convert PDF to 'PDF/A-1a' with multi PDF engines: convert PDF to 'PDF/A-1a' with PDFcpu: method not available; convert PDF to 'PDF/A-1a' with PDFtk: method not available; convert PDF to 'PDF/A-1a' with QPDF: method not available; convert PDF to 'PDF/A-1a' with unoconv: unoconv PDF: unix process error: wait for unix process: exit status 5","trace":"a4b50916-7779-4b54-b14a-883469dc98dc","remote_ip":"172.20.0.68","host":"converter:3000","uri":"/forms/chromium/convert/html","method":"POST","path":"/forms/chromium/convert/html","referer":"","user_agent":"ReactorNetty/1.0.17","status":500,"latency":2274155848,"latency_human":"2.274155848s","bytes_in":-1,"bytes_out":21}
```
Especially the part: `convert PDF to 'PDF/A-1a' with unoconv: unoconv PDF: unix process error: wait for unix process: exit status 5`

This happened using a gotenberg container, which was running for about 3 weeks in total. I guess the unoconv-service / listener failed / crashed and didn't recover? Either way, this shouldn't be the case I guess.

**Comment by PierreSchwang** [comment]
Small status update: Seems like this is no issue with long-term runs only, but happens on freshly restarted containers after 2 hours now as well :/

**Comment by gulien** [comment]
Hello @PierreSchwang,

Do you use the default value for `--uno-listener-restart-threshold` (i.e., 10)? You may try to reduce this value so that the long-running LibreOffice instances are restarted more often or go stateless with `--uno-listener-restart-threshold=0`.

**Comment by PierreSchwang** [comment]
> Hello @PierreSchwang,
> 
> Do you use the default value for `--uno-listener-restart-threshold` (i.e., 10)? You may try to reduce this value so that the long-running LibreOffice instances are restarted more often or go stateless with `--uno-listener-restart-threshold=0`.

Hi - No, we're running gotenberg with the default configuration. I guess using the stateless mode isn't that good of an idea, as this would increase the time for the conversion tremendously?

**Comment by gulien** [comment]
It depends on your use case 😊 for instance, batch conversions may actually finish quicker, because a long-running instance cannot handle concurrent conversions. Of course one could mitigate this by spawning multiple Gotenberg instances, but you get the idea.

**Comment by PierreSchwang** [comment]
The instance is only reachable using a proxy, but handles many single operations rather than few batches :/

**Comment by gulien** [comment]
What does the health check returns in your case? https://gotenberg.dev/docs/modules/api#health

**Comment by PierreSchwang** [comment]
Reproduced it on a different machine, health returns ok-results:
```json
{"status":"up","details":{"uno":{"status":"up","timestamp":"2022-10-05T20:08:46.2204444Z"}}}
```
Can be reproduced by running gotenberg without any additional configuration, and calling the chromium-convert enpoint (basically trigger the PDF/A-1a conversion):
```
curl \
--request POST 'http://localhost:3000/forms/chromium/convert/html' \
--form 'files=@"/path/to/index.html"' \
--form 'pdfFormat=@"PDF/A-1a"' \
-o my.pdf
```

At the beginning that will work flawlessly, after the container ran without any commands for a bit, that'll fail

**Comment by PierreSchwang** [comment]
Our current workaround is, by disabling the pdf-A conversion.

I reproduced it after about half an hour without commands, iirc

**Comment by ytilotti** [comment]
Hi @PierreSchwang, same problem here with version 7.6.1.

Process:
1. Up the instance of docker
2. Convert files with call /forms/libreoffice/convert
3. File pdf generated
4. Call /health, API up
5. Convert files with call /forms/libreoffice/convert
6. File pdf generated
7. Call /health, API up
8. Wait 10 minutes
9. Convert with call /forms/libreoffice/convert
10. Error `convert to PDF: unoconv PDF: unix process error: wait for unix process: exit status 5`
11. Call /health, API up
12. restart the instance
13. Convert with /forms/libreoffice/convert
14. File pdf generated

I don't understand. Ater few minutes without activity, the error appear without PDF/A-1a.

**Comment by gulien** [comment]
@ytilotti thanks for the details. Maybe the garbage collector is the culprit here, as it deletes files in `/tmp` after some time (i.e., if they are older than the allowed process time of a request). That being said, I've tested it many time and it did no cause any issues. Maybe something has changed with the latest version of LibreOffice.

In the meantime, could you go stateless? (`--uno-listener-restart-threshold=0`). I'm confident it will solve the issue, even if performance will drop.

**Comment by ytilotti** [comment]
@gulien I'm on already in testing `--uno-listener-restart-threshold=0`. Current in waiting period of 10 minutes :)

**Comment by gulien** [comment]
@ytilotti Your feedback was based on the stateless mode? If that's the case, it looks like the latest version of LibreOffice is the culprit...

**Comment by ytilotti** [comment]
My feedback was without stateless mode. After new test with stateless mode (`--uno-listener-restart-threshold=0`) it's working. More slow but working.

**Comment by cars10** [comment]
I also have this issue, i cannot convert epub files.

```
docker run --rm -p 3000:3000 gotenberg/gotenberg
curl -X POST 'http://localhost:3042/forms/libreoffice/convert' -F 'files=@"./Alices_Adventures_in_Wonderland.epub"' -o test.pdf
```

Results in the error message:

```
{"level":"error","ts":1667908795.8060749,"logger":"api","msg":"convert to PDF: unoconv PDF: unix process error: wait for unix process: exit status 5","trace":"0f13fdf3-7067-475b-b0ca-0e2b260300aa","remote_ip":"172.17.0.1","host":"localhost:3042","uri":"/forms/libreoffice/convert","method":"POST","path":"/forms/libreoffice/convert","referer":"","user_agent":"curl/7.86.0","status":500,"latency":136665529,"latency_human":"136.665529ms","bytes_in":886127,"bytes_out":21}
```

I have tried many epub files, the one in this example was taken from [here](https://filesamples.com/formats/epub).

**Comment by PhE** [comment]
The issue is still occuring with the latest 7.7.1

**Comment by gulien** [comment]
@PhE @cars10  @PierreSchwang @ytilotti  could you try the image `gulnap/gotenberg:libreofficefix`? It's using a downgraded version of LibreOffice (7.0.4) but with the Gotenberg 7.7.1 codebase.

**Comment by PierreSchwang** [comment]
> @PhE @cars10  @PierreSchwang @ytilotti  could you try the image `gulnap/gotenberg:libreofficefix`? It's using a downgraded version of LibreOffice (7.0.4) but with the Gotenberg 7.7.1 codebase.

Hi! After quick testing with multiple requests with a bit of a time difference (13:54, 14:24, 14:44) it seems to work so far. Do you know the culprit?
I'll try to test further but looks promising so far, thanks!

**Comment by gulien** [comment]
@PierreSchwang Looks like newer versions of LibreOffice are the culprit (or the related Debian packages).

**Comment by PierreSchwang** [comment]
> @PierreSchwang Looks like newer versions of LibreOffice are the culprit (or the related Debian packages).

Would be great to know the commit which introduced the (seemingly) bug. Was searching through gerrit for a bit but haven't found anything. Tho I'm not sure if it may be changed due to a mistake in implementing another feature.

**Comment by gulien** [comment]
> Would be great to know the commit which introduced the (seemingly) bug.

👍 yep, downgrading is indeed a temporary solution.

**Comment by gulien** [comment]
Release `7.7.2` is now available with the fix. Please open a new issue if the problem still persist 😄

**Comment by PierreSchwang** [comment]
> Release `7.7.2` is now available with the fix. Please open a new issue if the problem still persist 😄 

Deployed the update on our dev system and will monitor the behavior. I'll keep you updated!

**Comment by gulien** [comment]
Thanks @PierreSchwang!

**Comment by gulien** [comment]
Thanks @ytilotti!

**Comment by ytilotti** [comment]
An issue was sended to LibreOffice? 7.0.4 seems to be old.

**Comment by gulien** [comment]
Yes and no, the latest version available in Debian 11 is `7.4.3`, so no so far away from `7.0.4`. I wonder if it’s an issue with LibreOffice or the Debian package from backports channel.

**Comment by PierreSchwang** [comment]
Maybe related to https://bugs.documentfoundation.org/show_bug.cgi?id=146807
A crash could explain why the document is Null when attempting to load via unoconv. But I can't be sure sadly - If anyone want's to investigate further into that, I'd be glad. Will try to research and test further with 7.7.1 - maybe with "bisecting" there are new information to be found :/

**Comment by PierreSchwang** [comment]
I did a bit of research and came to the following result with a simple binary search:

Working | Not Working | Release Date
-- | -- | --
  | 7.7.1 (4.4.2.3) | 03.12.2022
  | 7.6.1 (7.4.1.2 40(Build:2)) | 03.10.2022
7.6.0 (7.3.5.2 30(Build:2)) |   | 01.09.2022
7.5.4 (7.3.5.2) |   |  
7.5.3 (7.3.4.1) |   | 07.06.2022

The version in brackets represents the used LibreOffice version in each gotenberg version.
That might explain the sudden issue on our end, as we used `gotenberg/gotenberg:7` as the image. I've created this issue on the 04th of October, and the update with the broken version of LO was released on the 03rd of October. Our CI/CD pipeline most certainly did a `docker-compose pull` which updated to the named version and explains the sudden error(s).

So, basically - the culprit is on of 1.185 commits between v7.4.1.2 and v7.3.5.2 (https://github.com/LibreOffice/core/compare/libreoffice-7.4.1.2...libreoffice-7.3.5.2) - fun to debug tbh
